CONTRIBUTING.md

```diff
@@ -1,81 +1,132 @@
-
-
-### Git history
-
-Pull Requests are squashed when merged. This means all commits in the branch will be squashed into one commit applied onto master, so branches and their PRs should stick to *one* topic only. If you think changes deserve separate commits, make separate PRs for each commit.
-
-This also means it's not possible to reuse one branch for multiple PRs; new PRs must either use entirely new branches, or else you could use `git reset --hard` on the current branch.
-
-
-### Allowed features
-
-Lc0 is still in early stages of development, and has not yet reached the point where we are ready to add small tweaks to add few points of a rating. Large code changes still happen, and having lots of small optimizations adds overhead to larger changes, slowing development.
-
-Therefore, as a rule, search algorithm tweaks that give a gain of less than ~20 Elo points are discouraged at this point. (This limit will gradually be lowered as Lc0 code matures, eventually to 0.0 Elo).
-
-
-#### Adding new command line flags/UCI parameters
-
-Only add new parameters if users can significantly (>20 Elo) benefit by tweaking it. We don't want to make every single constant configurable (or rather, users don't want to see hundreds of parameters which don't do anything).
-
-Try to minimize number of parameters that your feature introduces. If your feature introduces several parameters, every individual parameter should be significant (i.e. tweaking it with other fixes will give >20 Elo).
-
-
-#### Adding features for testing
-
-It is fine to temporarily commit a feature of unknown Elo gain so that people may test it. It's also fine to expose many parameters for the feature initially so that people can tune them. However, if the tweak doesn't prove to be significant, it should be removed after a few weeks.
-
+# Contributing to Leela Chess Zero (Lc0)
+
+Last updated: June 2025
+
+Thank you for your interest in contributing to LCZero! This document provides
+guidelines for contributing to the codebase.
+
+## Before you start
+
+* All contributors are encouraged to join our Discord server at
+  <https://lc0.org/chat>.
+* Refer to [README.md](README.md) for building and running instructions.
+  * The protobufs that are shared with the training code are located in a
+    separate repository. Don't forget to run `git submodule update --init` to
+    fetch them.
+* Familiarize yourself with the developer documentation at
+  <https://lczero.org/dev/>.
+
+We use the [Meson](https://mesonbuild.com/) build system.
+
+* In Linux, using `builddir/` is recommended for development
+  (`meson setup builddir/`), as VSCode recognizes it and all the development and
+  debugging tools work there (ask in Discord if you have issues).
+  * In the `builddir/`, run `ninja lc0` (which is faster than just `ninja`). Run
+    `ninja test` to run the tests.
+* In Windows, `meson setup build/debug` generates a Visual Studio solution that
```

Comment:
Are you sure about that? I think ninja is the default under windows as well. BTW I have started using ninja under windows as well, I find it easier to work with.

Reply:
That's how it was in 2018 when I still used windows. Maybe it's changed or there are several options.

At least `build.cmd` seems to invoke `msbuild` and not `ninja`.